Adrian AI Materials (#5)
Ada-Developers-Academy/independent-ai-topics: working-with-openai/open-ai-api-models.md

> ## Introduction to OpenAI API Models

Reviewer: Removed a hashtag to make the title an H1.

```suggestion
# Introduction to OpenAI API Models
```

| Embeddings | The embeddings model converts text to its numerical equivalent in order to measure the "relatedness" of text strings. | Search queries, clustering text by similarity | 
| Moderation | The moderation model can be used to check and see if content might be considered sensitive, unsafe or unsuitable for certain populations. | Content moderation |  

If you are interested in learning about other models that are no longer widely used, OpenAI provides a list [here](https://platform.openai.com/docs/deprecations)  

### !callout-info

## Where are OpenAI Models Used? 
As we will see in the next lesson on endpoints, these models are most commonly used as query parameters within our API requests to tell OpenAI which model we would like to use when processing a specific request. 

### !end-callout

### !callout-info
  
## OpenAI Models and Pricing
  
The OpenAI API is not a free service at this point in time, so each model has a different pricing structure that depends on the size of the request that is being made. Pricing information can be found [here](https://openai.com/api/pricing)  

### !end-callout

## Summary

> You should know have a basic understanding of OpenAI models and what they are used for. OpenAI provides models for everything from Text to Speech to Image Generation. If you are curious to know even more about OpenAI models, don't hesitate to do more research in the [OpenAI API Documentation](https://platform.openai.com/docs/guides/text-generation)

Reviewer: There's a typo in the first sentence, I took a look at restructuring that sentence and moving it to the middle. 

```suggestion
OpenAI provides models for everything from Text to Speech to Image Generation. We've also seen some ideas of applications that could incorporate the OpenAI models. If you are curious to know even more about the OpenAI models, don't hesitate to do more research in the [OpenAI API Documentation](https://platform.openai.com/docs/guides/text-generation)
```

> # Introduction to OpenAI API Models

Reviewer: To help differentiate lesson titles, even though we aren't doing an exhaustive dive I think it's okay if we shorten this to something like "OpenAI API's Models". 

```suggestion
# OpenAI API's Models
```

## Goals

Now that we have OpenAI set up on Postman, we can start taking a look under the hood at how OpenAI works and how we can integrate it into the programs we create. We will start by looking at the OpenAI API Models.

Our goals for this lesson are to:

> - Understand what an OpenAI Model is and how it is used
> - Recognize different OpenAI Models and how they are used
> - Understand the difference between different GPT models and how they are used.

Reviewer: This isn't necessary, but I think we can condense the goals a little while still capturing the intention and reducing some of the repeated phrasing at the end of the sentences.

```suggestion
- Recognize several commonly used OpenAI Models, their specific capabilities, and how they can be used.
- Understand some of the differences and limitations between the versions of GPT models.
```

| Embeddings | The embeddings model converts text to its numerical equivalent in order to measure the "relatedness" of text strings. | Search queries, clustering text by similarity | 
| Moderation | The moderation model can be used to check and see if content might be considered sensitive, unsafe or unsuitable for certain populations. | Content moderation |  

If you are interested in learning about other models that are no longer widely used, OpenAI provides a list [here](https://platform.openai.com/docs/deprecations)  

### !callout-info

## Where are OpenAI Models Used? 
As we will see in the next lesson on endpoints, these models are most commonly used as query parameters within our API requests to tell OpenAI which model we would like to use when processing a specific request. 

### !end-callout

### !callout-info
  
## OpenAI Models and Pricing
  
The OpenAI API is not a free service at this point in time, so each model has a different pricing structure that depends on the size of the request that is being made. Pricing information can be found [here](https://openai.com/api/pricing)  

### !end-callout

## Summary

> You should know have a basic understanding of OpenAI models and what they are used for. OpenAI provides models for everything from Text to Speech to Image Generation. If you are curious to know even more about OpenAI models, don't hesitate to do more research in the [OpenAI API Documentation](https://platform.openai.com/docs/guides/text-generation)

Reviewer: Missing punctuation to end the sentence.

```suggestion
You should know have a basic understanding of OpenAI models and what they are used for. OpenAI provides models for everything from Text to Speech to Image Generation. If you are curious to know even more about OpenAI models, don't hesitate to do more research in the [OpenAI API Documentation](https://platform.openai.com/docs/guides/text-generation)!
```
